src/App.tsx

```diff
@@ -15,36 +17,109 @@
+  visibleGoods: string[];
+};
+
+function sortGoods(
+  goods: string[],
+  sortField: string = '',
+  isReverse: boolean = false,
+): string[] {
+  const visibleGoods: string[] = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABET:
+        visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case SORT_FIELD_LENGTH:
+        visibleGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        throw new Error('Sort method is unknown!');
+    }
+  }
+
+  if (isReverse) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App: React.FC<Props> = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setReverse] = useState(false);
+  const visibleGoods = sortGoods(goodsFromServer, sortField, isReverse);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortField !== SORT_FIELD_ALPHABET,
+          })}
+          onClick={() => {
+            setSortField(SORT_FIELD_ALPHABET);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+          onClick={() => {
+            setSortField(SORT_FIELD_LENGTH);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button is-warning', {
+            'is-light': !isReverse,
+          })}
+          onClick={() => {
+            setReverse(!isReverse);
+          }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField !== SORT_FIELD_RESET || isReverse) && (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={() => {
+              setSortField(SORT_FIELD_RESET);
+              setReverse(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map(item => (
+          <li key={item} data-cy="Good">
+            {item}
+          </li>
+        ))}
```

Comment:
[CODE KNOWLEDGE] - When you're rendering a list, make sure to provide a unique `key` prop to each child. If there are items with the same name, using the item itself as a key could lead to rendering issues. Consider using an index or a unique identifier.